src/styles/main.scss

```diff
@@ -1,3 +1,62 @@
-body {
+$days: tue 1, wed 2, thu 3, fri 4, sat 5, sun 6;
+$cell-size: 100px;
+$calendar-padding: 10px;
+$calendar-gap: 1px;
+
+* {
+  box-sizing: border-box;
   margin: 0;
 }
+
+.calendar {
+  display: flex;
+  gap: $calendar-gap;
+  flex-wrap: wrap;
+  padding: $calendar-padding;
+  width: ($cell-size * 7) + ($calendar-gap * 6) + ($calendar-padding * 2);
+
+  &__day {
+    width: $cell-size;
+    height: $cell-size;
+    display: flex;
+    visibility: hidden;
+    text-align: center;
+    justify-content: center;
+    align-items: center;
+    background-color: #eee;
+    border: 1px solid black;
+
+    transition: transform 0.5s;
+    cursor: pointer;
+
+    &::before {
+      display: block;
+      font-size: 30px;
+      font-family: Arial, sans-serif;
+    }
+
+    &:hover{
+      transform: translateY(-20px);
+      background-color: pink;
+    }
+  }
+}
+
+@for $day from 28 through 31 {
+  .calendar--month-length-#{$day} {
+    @for $newDay from 1 through $day {
+      .calendar__day:nth-child(#{$newDay}) {
+        visibility: visible;
+        &::before {
+          content: $newDay + '';
+        }
+      }
+    }
+  }
+}
```

Comment:
This modifier breaks if we put a number less than 28 or more than 31, and it should work as follows, if we put more than 31 or less than 28, the modifier should simply not work and all dates should be visible.